src/hooks/useGraphQLClient.ts

```diff
@@ -1,32 +1,85 @@
-import { ApolloLink, createHttpLink, from, InMemoryCache, NormalizedCacheObject, Operation } from '@apollo/client';
+import {
+  ApolloLink,
+  createHttpLink,
+  from,
+  fromPromise,
+  InMemoryCache,
+  NormalizedCacheObject,
+  Operation,
+} from '@apollo/client';
 import { ApolloClient } from '@apollo/client/core/ApolloClient';
 import { setContext } from '@apollo/client/link/context';
 import { onError } from '@apollo/client/link/error';
 import { RetryLink } from '@apollo/client/link/retry';
+import { stat } from 'fs';
 import { useDispatch } from 'react-redux';
 import { env } from '../env';
 import { typePolicies } from '../graphql/cache/typePolicies';
 import { pushError } from '../reducers/error/actions';
+import { useAuthenticate } from './useAuthenticate';
 import { TOKEN_STORAGE_KEY } from './useAuthentication';
+import { updateStatus } from '../reducers/auth/actions';
+import { useCallback, useEffect } from 'react';
 
 const enableQueryDebug = !!(env && env?.REACT_APP_DEBUG_QUERY === 'true');
 
 export const useGraphQLClient = (graphQLEndpoint: string): ApolloClient<NormalizedCacheObject> => {
   const dispatch = useDispatch();
+  const { safeRefresh, status, isAuthenticated } = useAuthenticate();
   let token: string | null;
 
-  const errorLink = onError(({ graphQLErrors, networkError }) => {
+  const handleStorageChange = useCallback(
+    (e: StorageEvent) => {
+      if (e.key === TOKEN_STORAGE_KEY) {
+        if (e.newValue === null && e.newValue !== e.oldValue) {
+          // if (status === 'done') safeRefresh();
+        }
+      }
+    },
+    [safeRefresh, status]
+  );
+
+  useEffect(() => {
+    window.addEventListener('storage', handleStorageChange);
+    return () => {
+      window.removeEventListener('storage', handleStorageChange);
+    };
+  }, [handleStorageChange]);
+
+  const errorLink = onError(({ graphQLErrors, networkError, forward, operation }) => {
     let errors: Error[] = [];
 
     if (graphQLErrors) {
+      for (let err of graphQLErrors) {
+        switch (err?.extensions?.code) {
+          case 'UNAUTHENTICATED':
+            if (status === 'done') {
+              return fromPromise(
+                safeRefresh()
+                  .then(result => {
+                    dispatch(updateStatus('done'));
```

Comment:
The `safeRefresh` update status internally. Do you need to update it here? It could be discussed who's responsibility is to update the status.